common/app/com/commercetools/sunrise/common/sessions/SessionHandlerBase.java

```diff
@@ -0,0 +1,69 @@
+package com.commercetools.sunrise.common.sessions;
+
+import io.sphere.sdk.json.JsonException;
+import io.sphere.sdk.json.SphereJsonUtils;
+import org.slf4j.Logger;
+import play.libs.Json;
+import play.mvc.Http;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Optional;
+
+public abstract class SessionHandlerBase<T> implements SessionHandler<T> {
+
+    @Override
+    public void overwriteInSession(final Http.Session session, @Nullable final T value) {
+        if (value != null) {
+            overwriteRelatedValuesInSession(session, value);
+        } else {
+            removeRelatedValuesFromSession(session);
+        }
+    }
+
+    @Override
+    public void removeFromSession(final Http.Session session) {
+        removeRelatedValuesFromSession(session);
+    }
+
+    protected abstract void overwriteRelatedValuesInSession(final Http.Session session, final T value);
+
+    protected abstract void removeRelatedValuesFromSession(final Http.Session session);
+
+    protected Optional<String> findValue(final Http.Session session, final String key) {
```

Comment:
maybe `findValueByKey`? The name `findValue` sounds to me like you put the thing you search for as argument not the key.